### PR TITLE
Improve error logging.

### PR DESCRIPTION
**Pull request by kerneltime**
Include the actual key that failed, it is needed to be able to sort
through the service providers logs.

Signed-off-by: Ritesh H Shukla <ritesh@oneconcern.com>

### CODE REVIEW

File: pkg/cafs/writer.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func pFlush(
 			err = destination.Put(context.TODO(), pather(leafKey.String()), bytes.NewReader(buffer), storage.OverWrite)
 		}
 		if err != nil {
-			errC <- fmt.Errorf("write segment file: %v", err)
+			errC <- fmt.Errorf("write segment file: %s, err: %v", pather(leafKey.String()), err)
```
Comment (fredbi) [inline pkg/cafs/writer.go]: another way to put it using error stacking is: 
```suggestion
			errC <- fmt.Errorf("write segment file: %s, err: %w", pather(leafKey.String()), err)
```

File: pkg/cafs/writer.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func (w *fsWriter) flush(isLastNode bool) (int, error) {
 			err = w.store.Put(context.TODO(), w.pather(leafKey.String()), bytes.NewReader(w.buf[:w.offset]), storage.OverWrite)
 		}
 		if err != nil {
-			return 0, fmt.Errorf("write segment file: %v", err)
+			return 0, fmt.Errorf("write segment file: %s err:%v", w.pather(leafKey.String()), err)
```
Comment (fredbi) [inline pkg/cafs/writer.go]:
```suggestion
			return 0, fmt.Errorf("write segment file: %s err:%w", w.pather(leafKey.String()), err)
```

File: pkg/cafs/writer.go (hunk 1 above)
Comment (kerneltime) [reply pkg/cafs/writer.go]: done

File: pkg/cafs/writer.go (hunk 2 above)
Comment (kerneltime) [reply pkg/cafs/writer.go]: done

**Pull request merged**